### ISSUE TITLE
Dumping XML sitemap should dump for domain not for locales or webspaces

### ISSUE THREAD

**Issue by bixelsnl** [sulu/sulu#4094]
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.6.21
| Browser Version | Chrome 67.0.3396.99 (Official Build) (64-bit)

#### Actual Behavior

XML sitemap dumper generates `sitemap.xml` files for each webspace and each locale.

#### Expected Behavior

There should only be one sitemap: for the domain. Because as discussed with @alexander-schranz on Slack:

* sitemap.xml already contains links to all alternative languages, so there is no need for a separate `sitemap.xml` per locale
* there can be multiple webspaces on a single domain

@alexander-schranz has stated the only thing that would make sense is if it would output `var/sitemaps/https/domain.xml`

#### Steps to Reproduce

Given the following portal configuration:

```
    <portals>
        <portal>
            <name>mysite.nl</name>
            <key>mywebspace</key>

            <environments>
                <environment type="prod">
                    <urls>
                        <url>{host}/{localization}</url>
                        <url language="region">{host}</url>
                    </urls>
                </environment>
                <environment type="stage">
                    <urls>
                        <url>{host}/{localization}</url>
                        <url language="region">{host}</url>
                    </urls>
                </environment>
                <environment type="test">
                    <urls>
                        <url>{host}/{localization}</url>
                        <url language="region">{host}</url>
                    </urls>
                </environment>
                <environment type="dev">
                    <urls>
                        <url>{host}/{localization}</url>
                        <url language="region">{host}</url>
                    </urls>
                </environment>
            </environments>
        </portal>
    </portals>
```

Then running the command:

```
$ bin/websiteconsole sulu:website:dump-sitemap --https
```

Will output:

```
Start dumping "sitemap.xml" files:
 - mywebspace (nl_nl)
 - mywebspace (nl_be)
 - mywebspace (es_es)
 - mywebspace (fr_fr)
 - mywebspace (fr_lu)
 - mywebspace (fr_ch)
 - mywebspace (fr_be)
 - mywebspace (en_gb)
 - mywebspace (en_ie)
 - mywebspace (de_de)
 - mywebspace (de_at)
 - mywebspace (de_ch)
 - mywebspace (de_lu)
```

Resulting in these files:

```
var/sitemaps/https/mywebspace/nl_nl/mysite.nl/sitemap.xml
var/sitemaps/https/mywebspace/nl_be/mysite.nl/sitemap.xml
.. etc etc
```

#### Possible Solutions

* `DumpSitemapCommand` should dump only for the default localization, instead of iterating through all localizations
* `XmlSitemapDumper::getDumpPath()` should be adjusted to only dump to a domain.xml
* The call from `SitemapController` to `XmlSitemapDumper::getDumpPath()` also needs to be adjusted so it can find the cached file

**Comment by alexander-schranz** [comment]
Also currently the PageSitemapProvider works only for one portal: https://github.com/sulu/sulu/blob/50113b0cbd64c7c2de4c23bbeaca1420821c5257/src/Sulu/Bundle/WebsiteBundle/Sitemap/Provider/PagesSitemapProvider.php#L50-L57

**Comment by rs2487** [comment]
1.) @alexander-schranz why would that be better to give the domain? How can i decide which portal pages to load, wenn only the domain is given. It can be that i have the same domain for two portals  (under `/` and under `/subportal` e.g.). If i get only the domain i cannot decide which webspace specific contents i have to load for the sitemap.

2.) getMaxPage() must also return a webspace specific value.

**Comment by alexander-schranz** [comment]
@rs2487 because sitemaps have nothing todo with portals or webspaces. Sitemaps are always domain specific. The webspace manager or webspace collection should just create a easy way to get all portals / webspaces which are under a certain domain.

**Comment by danrot** [comment]
Fixed in #4777
